hypothesis-python/src/hypothesis/internal/conjecture/dfa/__init__.py

```diff
@@ -146,31 +203,88 @@ def count_strings(self, i, k):
         return sum(self.count_strings(j, k - 1) for _, j in self.transitions(i))
 
     @cached
-    def reachable(self, i):
-        """Returns the set of all states reachable
-        by traversing some non-empty string starting from
-        state i."""
-        reached = set()
+    def child_nodes(self, i):
+        seen = set()
+        result = []
+        for c, j in self.raw_transitions(i):
+            if j not in seen:
+                seen.add(j)
+                result.append(j)
+        return tuple(result)
 
-        queue = deque([i])
+    def is_dead(self, state):
+        """Returns True if no strings can be accepted
+        when starting from ``state``."""
+        return not self.is_live(state)
 
+    @cached
```

Comment:
Is this intentional? I haven't been able to figure out the benefit of having both `@cached` and an explicit cache lookup, given that `is_live` isn't recursive.

Reply:
It is not!